pkg/agent/core/state.go

```diff
@@ -995,6 +1106,11 @@ func (h PluginHandle) RequestSuccessful(now time.Time, resp api.PluginResponse)
 	// the process of moving the source of truth for ComputeUnit from the scheduler plugin to the
 	// autoscaler-agent.
 	h.s.Plugin.Permit = &resp.Permit
+	revsource.Propagate(now,
+		targetRevision,
+		&h.s.Plugin.CurrentRevision,
+		h.s.Config.ObservabilityCallbacks.PluginLatency,
+	)
```

Comment:
What happens if the request is successful but does not approve the requested resources?

Reply:
The propagation doesn't happen -> we don't measure the latency.

Reply:
> The propagation doesn't happen

Could you elaborate on that? AFAICT there is no special detection of "scheduler did not approve what we asked for", so we'll still call `revsource.Propagate`.

(do you mean that we won't make follow-up requests, so end-to-end latency isn't recorded until the request is approved? in that case: we'd still record plugin latency incorrectly though, right? and- that would mean that end-to-end latency records the _first_ change rather than the _full_ change — e.g. if we want to scale from 1 → 4 CU, but can only get approval for 3 CU first, latency of 1 → 3 will get recorded, but 1 → 4 won't, IIUC?)

Reply:
> Could you elaborate on that? AFAICT there is no special detection of "scheduler did not approve what we asked for", so we'll still call revsource.Propagate.

Sorry, I misread the question. Actually yes, the propagation happens as usual. 

> that would mean that end-to-end latency records the first change rather than the full change

Very good question. I wrote a test to verify this 8dcd87ac689e641384fe2cd5af5aece9fda3e3ad.

And then I changed the implementation so that sequence is considered as two upscalings: 
 - 1 -> 4 desired / 3 real
 - 3 -> 4

In 49bd61af4dcb1678bbaafaaca1bbea643868a3b9

Reply:
Nice!